src/main/java/io/confluent/connect/hdfs/HdfsSinkConnectorConfig.java

```diff
@@ -146,100 +198,223 @@
-      .define(HIVE_METASTORE_URIS_CONFIG, Type.STRING, HIVE_METASTORE_URIS_DEFAULT, Importance.HIGH, HIVE_METASTORE_URIS_DOC)
-      .define(SHUTDOWN_TIMEOUT_CONFIG, Type.LONG, SHUTDOWN_TIMEOUT_DEFAULT,
-              Importance.MEDIUM, SHUTDOWN_TIMEOUT_DOC)
-      .define(SCHEMA_COMPATIBILITY_CONFIG, Type.STRING, SCHEMA_COMPATIBILITY_DEFAULT,
-              Importance.HIGH, SCHEMA_COMPATIBILITY_DOC)
-      .define(PARTITIONER_CLASS_CONFIG, Type.STRING, PARTITIONER_CLASS_DEFAULT, Importance.HIGH, PARTITIONER_CLASS_DOC)
-      .define(PARTITION_FIELD_NAME_CONFIG, Type.STRING, PARTITION_FIELD_NAME_DEFAULT, Importance.MEDIUM,
-              PARTITION_FIELD_NAME_DOC)
-      .define(PARTITION_DURATION_MS_CONFIG, Type.LONG, PARTITION_DURATION_MS_DEFAULT, Importance.MEDIUM,
-              PARTITION_DURATION_MS_DOC)
-      .define(PATH_FORMAT_CONFIG, Type.STRING, PATH_FORMAT_DEFAULT, Importance.MEDIUM,
-              PATH_FORMAT_DOC)
-      .define(LOCALE_CONFIG, Type.STRING, LOCALE_DEFAULT, Importance.MEDIUM, LOCALE_DOC)
-      .define(TIMEZONE_CONFIG, Type.STRING, TIMEZONE_DEFAULT, Importance.MEDIUM, TIMEZONE_DOC)
-      .define(FILENAME_OFFSET_ZERO_PAD_WIDTH_CONFIG, Type.INT,
-              FILENAME_OFFSET_ZERO_PAD_WIDTH_DEFAULT, ConfigDef.Range.atLeast(0),
-              Importance.LOW, FILENAME_OFFSET_ZERO_PAD_WIDTH_DOC)
-      .define(HDFS_AUTHENTICATION_KERBEROS_CONFIG, Type.BOOLEAN,
-              HDFS_AUTHENTICATION_KERBEROS_DEFAULT, Importance.HIGH,
-              HDFS_AUTHENTICATION_KERBEROS_DOC)
-      .define(CONNECT_HDFS_PRINCIPAL_CONFIG, Type.STRING, CONNECT_HDFS_PRINCIPAL_DEFAULT,
-              Importance.HIGH, CONNECT_HDFS_PRINCIPAL_DOC)
-      .define(CONNECT_HDFS_KEYTAB_CONFIG, Type.STRING, CONNECT_HDFS_KEYTAB_DEFAULT,
-              Importance.HIGH, CONNECT_HDFS_KEYTAB_DOC)
-      .define(HDFS_NAMENODE_PRINCIPAL_CONFIG, Type.STRING, HDFS_NAMENODE_PRINCIPAL_DEFAULT,
-              Importance.HIGH, HDFS_NAMENODE_PRINCIPAL_DOC)
-      .define(KERBEROS_TICKET_RENEW_PERIOD_MS_CONFIG, Type.LONG,
-              KERBEROS_TICKET_RENEW_PERIOD_MS_DEFAULT, Importance.LOW,
-              KERBEROS_TICKET_RENEW_PERIOD_MS_DOC);
+  private static class SchemaCompatibilityRecommender extends BooleanParentRecommender {
+    
+    public SchemaCompatibilityRecommender() {
+      super(HIVE_INTEGRATION_CONFIG);
+    }
+      
+    @Override
+    public List<Object> validValues(String name, Map<String, String> connectorConfigs) {
+      boolean hiveIntegration = Boolean.parseBoolean(connectorConfigs.get(parentConfigName));
+      if (hiveIntegration) {
+        return Arrays.<Object>asList("BACKWARD", "FORWARD", "FULL");
+      } else {
+        return Arrays.<Object>asList("NONE", "BACKWARD", "FORWARD", "FULL");
+      }
+    }
+
+    @Override
+    public boolean visible(String name, Map<String, String> connectorConfigs) {
+      return true;
+    }
+  }
+  
+  private static class BooleanParentRecommender implements ConfigDef.Recommender {
```

Comment:
This one definitely seems like a very reusable implementation. Should this go somewhere that is easier to share? We can always do this as a refactoring, but worth thinking about what we can reuse.